src/app/beer_garden/db/mongo/util.py

```diff
@@ -297,7 +242,7 @@ def check_indexes(document_class):
             db = get_db()
             db[document_class.__name__.lower()].drop_indexes()
             logger.warning("Dropped indexes for %s collection", document_class.__name__)
-        except OperationFailure:
+        except OperationError:
```

Comment:
This is probably true of the earlier `ensure_indexes()` call too, but switching from `pymongo.errors.OperationFailure` to `mongoengine.errors.OperationError` here is not going to work.  The `drop_indexes()` call is a pymongo one, so there's no situation where it's going to raise a mongoengine errror.

This might be the point where the goal of removing the `pymongo` dependency completely becomes a fool's errand.  Mongoengine doesn't expose a way drop indexes and instead provides these mechanism to get at the pymongo objects do it through those.

Others may have opinions or ideas, but we may need to accept the reality that using mongoengine means interacting with pymongo directly for some things.